test/CarouselSpec.js

```diff
@@ -100,6 +100,24 @@ describe('<Carousel>', () => {
 
       wrapper.find('.carousel-indicators li').last().simulate('click');
     });
+
+    it(`should allow refs to be attached and expose next, prev functions`, () => {
+      let inputRef = React.createRef();
+      const onSelectSpy = sinon.spy();
+      const wrapper = mount(
+        <Carousel ref={inputRef} onSelect={onSelectSpy}>
+          {items}
+        </Carousel>,
+      );
+      expect(inputRef.current).to.have.property('next');
+      expect(inputRef.current).to.have.property('prev');
+      expect(inputRef.current).to.have.property('element');
+      inputRef.current.next();
+      expect(onSelectSpy).to.have.been.calledOnce;
+      inputRef.current.prev();
+      expect(onSelectSpy).to.have.been.calledOnce;
+      wrapper.unmount();
+    });
```

Comment:
This test is being run inside the `['onSlide', 'onSlid'].forEach` block. Should be outside since it doesn't depend on it

Also, since you're invoking prev and next, the select spy should be called twice.

Reply:
updated for all